### PR TITLE
Fixes #22796 - fact importing telemetry

### PR DESCRIPTION
**Pull request by lzap**
SSIA

This patch is an example on how to add telemetry.

It removes publication of Rails notifications in the importer, it was
unused and we now have a better way of publishing that into telemetry
stack.

### CODE REVIEW

File: app/models/host/base.rb [hunk 1]
```diff
@@ -195,19 +201,21 @@ def set_interfaces(parser)
...
 
       ipmi = parser.ipmi_interface
       if ipmi.present?
         existing = self.interfaces.where(:mac => ipmi[:macaddress], :type => Nic::BMC.name).first
         iface = existing || Nic::BMC.new(:managed => false)
         iface.provider ||= 'IPMI'
-        set_interface(ipmi, 'ipmi', iface)
+        changed_count += 1 if set_interface(ipmi, 'ipmi', iface)
       end
+      telemetry_increment_counter(:importer_facts_count_interfaces, changed_count, type: parser.class_name_humanized)
```
Comment (dLobatog) [inline app/models/host/base.rb:218]: This line is unexpected in some tests that use a mocked parser, could you check that out?

**Review by dLobatog — CHANGES_REQUESTED** [review]
@lzap Thanks, this is much cleane! The failing tests are only failing because in `test/models/host_test.rb` you need to also mock the `class_name_humanized` method for the stubbed parsers in  `setup_host_with_nic_parser` and  ` setup_host_with_ipmi_parser`. 

:+1: to merge after that if tests are green

File: app/models/host/base.rb (hunk 1 above)
Comment (lzap) [reply app/models/host/base.rb:218]: Don't understand this one?

**Comment by lzap** [comment]
Thanks rebased.

**Comment by dLobatog** [comment]
[test foreman] 

I'll fix the hound spacing issues if tests pass

**Comment by dLobatog** [comment]
Tests pass, the failure in MySQL is being addressed on https://github.com/theforeman/foreman/pull/5325/files 
I'll fix the rubocop warning & merge. Thanks @lzap :medal_military:

**Review by dLobatog — APPROVED** [review]
Doh, I forgot 'develop' is now protected.. 

```
remote: error: GH006: Protected branch update failed for refs/heads/develop.
remote: error: At least 1 approving review is required by reviewers with write access.
```

Please just address the hound issues & I'll merge right away. Thanks !

**Comment by san7ket** [comment]
Can successfully list the facts telemetry
```
fm_rails_importer_facts_count_interfaces{type="puppet_fact_parser"} 1
# HELP fm_rails_importer_facts_count_processed Number of facts processed (added, updated, deleted) per importer type
# TYPE fm_rails_importer_facts_count_processed counter
fm_rails_importer_facts_count_processed{action="added",type="puppet_fact_name"} 0
fm_rails_importer_facts_count_processed{action="deleted",type="puppet_fact_name"} 11
fm_rails_importer_facts_count_processed{action="updated",type="puppet_fact_name"} 26
# HELP fm_rails_importer_facts_import_duration Duration of fact import (ms) per importer type
# TYPE fm_rails_importer_facts_import_duration histogram
fm_rails_importer_facts_import_duration_bucket{type="",le="10"} 3
```

**Comment by lzap** [comment]
Rebased.

**Comment by mmoll** [comment]
merged, díky @lzap!

**Pull request merged**